discussion/avoiding-mutable-aliases-to-hardware.tex

```diff
@@ -75,8 +78,8 @@ \subsection{Limitations with the Bindings}
     \item The peripherals are owned by a variable, and access to these peripherals can be controlled by {\rust}'s borrow-checker.
 \end{enumerate}
 
-Point 1) is indeed a very interesting feature of {\zinc}, but it does not help us to solve the problem that we are discussing in this section.
-Point 2) can help us to make sure that all peripheral access goes through the \code{run\_args} structure.
+Point 1) is indeed a very interesting feature of {\zinc}, but it does not help us solve the problem that we are discussing in this section.
+Point 2) can help us ensure that all peripheral access goes through the \code{run\_args} structure.
 However, this structure is only passed to the program's main loop, which has a different environment than the scopes that are defined by the interrupt handlers.
```

Comment:
` ... which has a different execution context than the interrupt handlers. This prevents {\rust} from statically analysing the problem.```